### PR TITLE
Add options for generated enum format.

### PR DESCRIPTION
**Pull request by crc1021**
I too ran into this issue https://github.com/eerimoq/pbtools/issues/9.  Here is one way to handle that.
I addition, I usually use all upper case for enum constants, so I added an option for that as well.

### CODE REVIEW

File: pbtools/c_source/__init__.py [hunk 1]
```diff
@@ -1307,17 +1334,18 @@ def generate(self):
...
 
-    return Generator(namespace, parsed, header_name).generate()
+    return Generator(namespace, parsed, header_name, gen_opts).generate()
 
 
 def generate_files(infiles,
                    import_paths=None,
-                   output_directory='.'):
+                   output_directory='.',
+                   gen_opts=None):
```
Comment (eerimoq) [inline pbtools/c_source/__init__.py:1348]: I'm fine with naming it `options` instead of `gen_opts`.

File: pbtools/c_source/__init__.py [hunk 2]
```diff
@@ -682,15 +691,24 @@
 '''
 
 
+class GeneratorOptions:
+
+    def __init__(self):
+        self.enum_prefix = 2
+        self.enum_upper = False
```
Comment (eerimoq) [inline pbtools/c_source/__init__.py:698]: I prefer passing the member values as arguments to `__init__`.

File: pbtools/subparsers/generate_c_source.py [hunk 3]
```diff
@@ -27,4 +32,13 @@ def add_subparser(subparsers):
         'infiles',
         nargs='+',
         help='Input protobuf file(s).')
+    subparser.add_argument(
+        '--enum-prefix',
+        default=2,
+        type=int,
+        help='set generated enum prefix, 0: none, 1: namespace, else: namespace and type')
```
Comment (eerimoq) [inline pbtools/subparsers/generate_c_source.py:39]: Should make it a `choice` with choices as strings.

Something like:

```python
choices=['namespace-and-type', 'none', 'namespace']
```

File: pbtools/c_source/__init__.py (hunk 1 above)
Comment (crc1021) [reply pbtools/c_source/__init__.py:1352]: change have been made

**Comment by eerimoq** [comment]
I'm tempted to never add the enum name itself as a prefix, and just rely on that enum field names are according to protobuf recommendation. That is, we can remove the --enum-prefix argument.

**Comment by crc1021** [comment]
> I'm tempted to never add the enum name itself as a prefix, and just rely on that enum field names are according to protobuf recommendation. That is, we can remove the --enum-prefix argument.

Are you saying that the format should not even include namespace?
`{field.name_snake_case}_e = {field.field_number}`
or keep namespace?
`{enum.namespace_snake_case}_{field.name_snake_case}_e = {field.field_number}`

**Comment by eerimoq** [comment]
The latter. It should include everything except the name of the enum.

**Comment by eerimoq** [comment]
I'll fix tests and clean it up.

**Pull request merged**